src/url_search_params.rs

```diff
@@ -0,0 +1,483 @@
+use crate::{ffi, ParseUrlError};
+
+pub struct UrlSearchParams(*mut ffi::ada_url_search_params);
+
+impl Drop for UrlSearchParams {
+    fn drop(&mut self) {
+        unsafe { ffi::ada_free_search_params(self.0) }
+    }
+}
+
+impl UrlSearchParams {
+    /// Parses an return a UrlSearchParams struct.
+    ///
+    /// ```
+    /// use ada_url::UrlSearchParams;
+    /// let params = UrlSearchParams::parse("a=1&b=2")
+    ///     .expect("This is a valid UrlSearchParams. Should have parsed it.");
+    /// assert_eq!(params.get("a"), Some("1"));
+    /// assert_eq!(params.get("b"), Some("2"));
+    /// ```
+    pub fn parse<Input>(input: Input) -> Result<Self, ParseUrlError<Input>>
+    where
+        Input: AsRef<str>,
+    {
+        Ok(Self(unsafe {
+            ffi::ada_parse_search_params(input.as_ref().as_ptr().cast(), input.as_ref().len())
+        }))
+    }
+
+    /// Returns the size of the UrlSearchParams struct.
```

Comment:
```suggestion
    /// Returns the length of the [`UrlSearchParams`].
```

This is a bit confusing. According to the doc example, this is not the length of the underlying string. Is it the number of keys, or the number of values then (as this is essentially a multi-value map)?

Reply:
I think it's best to pretend this object is a thin wrapper around `BTreeMap<String, Vec<String>>` and shape the API according to that. From that it becomes obvious what len, IntoIterator etc should be. I might be biased because I'm still expecting a lot of things to work like werkzeug's MultiDict.

I would say that `x.into_iter().count() == x.len()` as well.

Reply:
Agreed! I've updated the implementation.